desktop/models/sale.coffee

```diff
@@ -189,8 +189,16 @@ module.exports = class Sale extends Backbone.Model
       "Auction opens #{@zone(@get('start_at')).format(timeFormat)}"
     else if !@get('is_sale')
       "Closes #{@zone(@get('end_at')).format(timeFormat)}"
-    else
+    else if @get('is_auction')
       "Bidding closes #{@zone(@get('end_at')).format(timeFormat)}"
+    else
+      ""
+
+    # Guard against missing data
+    if label.includes('Invalid')
```

Comment:
Feels a little brittle to be checking for `Invalid` in the text here. Could there be a condition to check against `@get('end_at')` earlier on?

Reply:
@sweir27 - I agree it appears somewhat brittle, but it is a good catch-all given the existing complexity of the model and how moment.js will always return this string for bad data. `upcomingLabel()` also deals with multiple date fields such as `start_at`, `end_at`, `live_start_at`, etc.